### PR TITLE
[CLI] ErrorAndExit deprecated

### PR DESCRIPTION
**Pull request by samkitshah1262**
Towards Issue #6317 

ErrorAndExit function is deprecated.  Now every error is propagated instead of exiting directly.

For best practices

**How did you test it?**

**Potential risks**

### CODE REVIEW

File: tools/cli/admin_cluster_commands.go [hunk 1]
```diff
@@ -83,8 +92,11 @@ func AdminDescribeCluster(c *cli.Context) error {
 		return err
 	}
 
-	ctx, cancel := newContext(c)
+	ctx, cancel, err := newContext(c)
 	defer cancel()
+	if err != nil {
+		return commoncli.Problem("Error in creating :", err)
```
Comment (shijiesheng) [inline tools/cli/admin_cluster_commands.go:98]: nit: context

File: tools/cli/admin_cluster_commands.go [hunk 2]
```diff
@@ -59,7 +65,10 @@ func AdminAddSearchAttribute(c *cli.Context) error {
 	if err != nil {
 		return err
 	}
-	ctx, cancel := newContext(c)
+	ctx, cancel, err := newContext(c)
+	if err != nil {
+		return commoncli.Problem("Error in creating context: %v", err)
```
Comment (shijiesheng) [inline tools/cli/admin_cluster_commands.go:70]: It should be 
```suggestion
		return commoncli.Problem("Error in creating context", err)
```

File: tools/cli/admin_commands.go [hunk 3]
```diff
@@ -52,13 +52,19 @@ func AdminShowWorkflow(c *cli.Context) error {
...
+		return commoncli.Problem("Error in creating context: %v", err)
+	}
 	serializer := persistence.NewPayloadSerializer()
 	var history []*persistence.DataBlob
 	if len(tid) != 0 {
 		thriftrwEncoder := codec.NewThriftRWEncoder()
-		histV2 := initializeHistoryManager(c)
+		histV2, err := initializeHistoryManager(c)
+		if err != nil {
+			return commoncli.Problem("Error in Admin delete WF: ", err)
```
Comment (shijiesheng) [inline tools/cli/admin_commands.go:66]: ditto

File: tools/cli/admin_commands.go [hunk 4]
```diff
@@ -170,13 +176,21 @@ func describeMutableState(c *cli.Context) (*types.AdminDescribeWorkflowExecution
...
+		return nil, commoncli.Problem("Required flag not found", err)
+	}
 	rid := c.String(FlagRunID)
 
-	ctx, cancel := newContext(c)
+	ctx, cancel, err := newContext(c)
 	defer cancel()
-
+	if err != nil {
+		return nil, commoncli.Problem("Error in creating context: %v", err)
```
Comment (shijiesheng) [inline tools/cli/admin_commands.go:192]: ditto

File: tools/cli/utils.go [hunk 5]
```diff
@@ -525,35 +525,35 @@ func printError(msg string, err error) {
 }
 
 // ErrorAndExit print easy to understand error msg first then error detail in a new line
-func ErrorAndExit(msg string, err error) {
-	printError(msg, err)
-	osExit(1)
-}
+// func ErrorAndExit(msg string, err error) {
```
Comment (davidporter-id-au) [inline tools/cli/utils.go:528]: probably can remove?

File: tools/cli/utils.go [hunk 6]
```diff
@@ -574,30 +574,30 @@ func convertTime(unixNano int64, onlyTime bool) string {
...
+		return resultValue, nil
 	}
 
 	// treat as time range format
 	parsedTime, err = parseTimeRange(timeStr)
 	if err != nil {
-		ErrorAndExit(fmt.Sprintf("Cannot parse time '%s', use UTC format '2006-01-02T15:04:05Z', "+
-			"time range or raw UnixNano directly. See help for more details.", timeStr), err)
+		return 0, fmt.Errorf("Cannot parse time '%s', use UTC format '2006-01-02T15:04:05Z', "+
+			"time range or raw UnixNano directly. See help for more details: %v", timeStr, err)
```
Comment (davidporter-id-au) [inline tools/cli/utils.go:598]: minor nit: prefer %w where possible due to the benefits of error wrappign

File: tools/cli/utils.go [hunk 7]
```diff
@@ -730,49 +730,69 @@ func processJWTFlags(ctx context.Context, cliCtx *cli.Context) context.Context {
 	} else if path != "" {
 		token, err = createJWT(path)
 		if err != nil {
-			ErrorAndExit("Error creating JWT token", err)
+			return nil, fmt.Errorf("error creating JWT token: %v", err)
```
Comment (davidporter-id-au) [inline tools/cli/utils.go:733]: minor nit: prefer %w for error wrapping

**Review by davidporter-id-au — COMMENTED** [review]
minor nit about that small chunk of commented code, otherwise LGTM, TY!

**Review by shijiesheng — APPROVED** [review]

**Pull request merged**